flink-runtime/src/main/java/org/apache/flink/runtime/blob/BlobClient.java

```diff
@@ -415,13 +393,17 @@ private BlobKey putBuffer(@Nullable JobID jobId, byte[] value, int offset, int l
 	 * 		the ID of the job the BLOB belongs to (or <tt>null</tt> if job-unrelated)
 	 * @param inputStream
 	 * 		the input stream to read the data from
+	 * @param permanentBlob
+	 * 		whether the BLOB is permanent (<tt>true</tt>) or transient (<tt>false</tt>)
 	 *
 	 * @return the computed BLOB key of the uploaded BLOB
 	 *
 	 * @throws IOException
 	 * 		thrown if an I/O error occurs while uploading the data to the BLOB server
 	 */
-	private BlobKey putInputStream(@Nullable JobID jobId, InputStream inputStream) throws IOException {
+	BlobKey putInputStream(@Nullable JobID jobId, InputStream inputStream, boolean permanentBlob)
```

Comment:
I'm not sure, it's worth it since this method is about to become private to `TransientBlobCache` once the BLOB upload is not necessary from the client anymore, i.e. once we upload the job in a single (HTTP) operation